### PR TITLE
ELF: add EM_AMDGPU entry

### PR DESCRIPTION
**Pull request by littlewu2508**
With this, portage is able to detect AMDGPU machine code elf object.

Closes: https://bugs.gentoo.org/795825
Signed-off-by: Yiyang Wu <xgreenlandforwyy@gmail.com>

### CODE REVIEW

File: lib/portage/dep/soname/multilib_category.py [hunk 1]
```diff
@@ -70,6 +70,7 @@
     E_MIPS_ABI_EABI64,
     E_MIPS_ABI_O32,
     E_MIPS_ABI_O64,
+    EM_AMDGPU,
```
Comment (Arfrever) [inline lib/portage/dep/soname/multilib_category.py:73]: Please add after `EM_ALPHA`.
Similar sorting below.

Comment (littlewu2508) [reply lib/portage/dep/soname/multilib_category.py]: Thanks, adjusted

File: lib/portage/dep/soname/multilib_category.py [hunk 2]
```diff
@@ -77,6 +78,7 @@
     EM_68K: "m68k",
     EM_AARCH64: "arm",
     EM_ALPHA: "alpha",
+    EM_AMDGPU: "AMDGPU",
```
Comment (Arfrever) [inline lib/portage/dep/soname/multilib_category.py:81]: Other values in this dictionary are lower-case, and this one probably also should be:
`EM_AMDGPU: "amdgpu",`

Comment (littlewu2508) [reply lib/portage/dep/soname/multilib_category.py]: Fixed

**Comment by thesamesam** [comment]
Could you rebase? Thanks!

**Comment by thesamesam** [comment]
Can you update NEWS please too? Thanks.

**Comment by littlewu2508** [comment]
> Can you update NEWS please too? Thanks.

Sure. My first contribution to portage, not sure if classifying this change as new feature correct or not.

**Comment by thesamesam** [comment]
I'd say it is. Thank you!

**Pull request closed without merging**